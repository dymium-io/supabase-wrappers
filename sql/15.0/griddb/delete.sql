--
-- GridDB
-- DELETE
--
\set ECHO none
\ir sql/configs/griddb_parameters.conf
\set ECHO all
\i sql/15.0/delete.sql
