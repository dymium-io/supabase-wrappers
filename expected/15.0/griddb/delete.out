--
-- GridDB
-- DELETE
--
\set ECHO none
\i sql/15.0/delete.sql
--Testcase 1:
CREATE EXTENSION :DB_EXTENSIONNAME;
--Testcase 2:
CREATE SERVER :DB_SERVERNAME FOREIGN DATA WRAPPER :DB_EXTENSIONNAME OPTIONS(
drivername :DB_DRIVERNAME,
url :DB_URL,
querytimeout '15',
jarfile :DB_DRIVERPATH,
maxheapsize '600'
);
--Testcase 3:
CREATE USER MAPPING FOR public SERVER :DB_SERVERNAME OPTIONS(username :DB_USER,password :DB_PASS);
--Testcase 4:
CREATE FOREIGN TABLE delete_test (
    id serial OPTIONS (key 'true'),
    a INT,
    b text
) SERVER :DB_SERVERNAME OPTIONS ( table_name 'delete_test');
INSERT INTO delete_test (a) VALUES (10);
INSERT INTO delete_test (a, b) VALUES (50, repeat('x', 10000));
INSERT INTO delete_test (a) VALUES (100);
-- allow an alias to be specified for DELETE's target table
--Testcase 5:
EXPLAIN VERBOSE DELETE FROM delete_test AS dt WHERE dt.a > 75;
                                     QUERY PLAN                                      
-------------------------------------------------------------------------------------
 Delete on public.delete_test dt  (cost=100.00..166.06 rows=0 width=0)
   Remote SQL: DELETE FROM delete_test WHERE id=?
   ->  Foreign Scan on public.delete_test dt  (cost=100.00..166.06 rows=975 width=4)
         Output: id
         Remote SQL: SELECT id FROM delete_test WHERE ((a > 75))
(5 rows)

--Testcase 6:
DELETE FROM delete_test AS dt WHERE dt.a > 75;
-- if an alias is specified, don't allow the original table name
-- to be referenced
--Testcase 7:
EXPLAIN VERBOSE DELETE FROM delete_test dt WHERE delete_test.a > 25;
psql:sql/15.0/delete.sql:34: ERROR:  invalid reference to FROM-clause entry for table "delete_test"
LINE 1: EXPLAIN VERBOSE DELETE FROM delete_test dt WHERE delete_test...
                                                         ^
HINT:  Perhaps you meant to reference the table alias "dt".
--Testcase 8:
DELETE FROM delete_test dt WHERE delete_test.a > 25;
psql:sql/15.0/delete.sql:36: ERROR:  invalid reference to FROM-clause entry for table "delete_test"
LINE 1: DELETE FROM delete_test dt WHERE delete_test.a > 25;
                                         ^
HINT:  Perhaps you meant to reference the table alias "dt".
--Testcase 10:
SELECT id, a, char_length(b) FROM delete_test;
 id | a  | char_length 
----+----+-------------
  1 | 10 |            
  2 | 50 |       10000
(2 rows)

-- delete a row with a TOASTed value
--Testcase 11:
EXPLAIN VERBOSE DELETE FROM delete_test WHERE a > 25;
                                    QUERY PLAN                                    
----------------------------------------------------------------------------------
 Delete on public.delete_test  (cost=100.00..166.06 rows=0 width=0)
   Remote SQL: DELETE FROM delete_test WHERE id=?
   ->  Foreign Scan on public.delete_test  (cost=100.00..166.06 rows=975 width=4)
         Output: id
         Remote SQL: SELECT id FROM delete_test WHERE ((a > 25))
(5 rows)

--Testcase 12:
DELETE FROM delete_test WHERE a > 25;
--Testcase 13:
SELECT id, a, char_length(b) FROM delete_test;
 id | a  | char_length 
----+----+-------------
  1 | 10 |            
(1 row)

--Testcase 15:
DROP FOREIGN TABLE delete_test;
--Testcase 16:
DROP SERVER :DB_SERVERNAME CASCADE;
psql:sql/15.0/delete.sql:52: NOTICE:  drop cascades to user mapping for public on server griddb_server
--Testcase 17:
DROP EXTENSION :DB_EXTENSIONNAME CASCADE;
